x-pack/packages/kbn-slo-schema/src/rest_specs/slo.ts

```diff
@@ -123,6 +133,8 @@ const sloWithSummaryResponseSchema = t.intersection([
   t.type({ summary: summarySchema }),
 ]);
 
+const sloGroupsResponseSchema = t.record(t.string, t.number);
```

Comment:
Since this is behind a feature flag, it's fine to start with the result being a `Record<group, number of element in group>` but when we introduce more stats in the response, we'll have to change the results to either an array or a Record<group, Stat Object>.

Reply:
Yep definitely the type will change when we introduce a) the stats and b) I guess when we introduce more group by options. So it will be more like:

```
{
"by_tags": {
   "test": {
       total: 5,
       worstValue: ...
       ...
     },
  },
 "by_status": {
     "healthy": StatsObject,
     "violated": StatsObject
  }
},
"by_sli_indicator": {
...
}

```

Is it fine to keep it like this for now? Once this is merged, I will work on the summary stats, so this change very soon

Reply:
I'm not sure returning all the group options (by tags, by indicator, by sttus_ at once is ideal since you'll need to handle pagination within each, and also based on the API input parameter, we explicitly ask for the group we want, i.e. `groupBy: groupBySchema`
I'd imagine the API being something like:

```
> Request
GET /api/observability/slos/groups?page=2&perPage=10&groupBy=tags
< Response
{
  total: 13
  page: 2
  perPage: 10,
  results: [
    { group: "tag-value-ccc", stats: { healthy: 82, violated: 1, degrading: 3, total: 86  } },
    { group: "tag-value-aaa", stats: { healthy: 4, violated: 0, degrading: 0, total: 4  } },
    { group: "tag-value-bbb", stats: { healthy: 0, violated: 10, degrading: 0, total: 10  } },
  ]
}

```

Reply:
Ok so in case of grouping by status for example, you say we would still have the same response like this:

```
> Request
GET /api/observability/slos/groups?page=2&perPage=10&groupBy=status
< Response
{
  total: 13
  page: 2
  perPage: 10,
  results: [
    { group: "healthy", stats: { healthy: 82, violated: 1, degrading: 3, total: 86  } },
    { group: "violated", stats: { healthy: 4, violated: 0, degrading: 0, total: 4  } },
  ]
}
```

Reply:
Yep indeed the API accepts `groupBySchema`. At the moment I don't make use of it, but eventually we will as part of https://github.com/elastic/kibana/issues/174329

Reply:
Correct. For status groups the stats are not very useful (except the total) but as we add more stats they will make sense.